carla/recourse_methods/autoencoder/models.py

```diff
@@ -0,0 +1,150 @@
+import os
+from typing import Callable, List, Optional
+
+import numpy as np
+from keras import backend as K
+from keras.layers import Dense, Input
+from keras.models import Model, Sequential
+
+
+def layers_valid(layers: List) -> bool:
+    """
+    Checks if the layers parameter has at least minimal requirements
+
+    Returns
+    -------
+    bool
+    """
+    if len(layers) < 2:
+        return False
+
+    for layer in layers:
+        if layer <= 0:
+            return False
+
+    return True
+
+
+class Autoencoder:
+    def __init__(
+        self,
+        layers: List,
+        data_name: str,
+        optimizer: str = "rmsprop",
+        loss: Optional[Callable] = None,
+    ) -> None:
+        """
+        Defines the structure of the autoencoder
+
+        Parameters
+        ----------
+        layers : list(int > 0)
+            Depending on the position and number elements, it determines the number and width of layers in the form of
+            [input_layer, hidden_layer_1, ...., hidden_layer_n, latent_dimension]
+        data_name : str
+            Name of the dataset. Is used for saving model.
+        loss: Callable, optional
+            Loss function for autoencoder model. Default is Binary Cross Entropy.
+        optimizer: str, optional
+            Optimizer which is used to train autoencoder model. See keras optimizer.
+        """
+        if layers_valid(layers):
+            self._layers = layers
+        else:
+            raise ValueError(
+                "Number of layers have to be at least 2 (input and latent space), and number of neurons bigger than 0"
+            )
+
+        if loss is None:
+            self._loss = lambda y_true, y_pred: K.sum(
```

Comment:
Nice use of a anonymous function, but i think in this case it would be nice to actually give it a name and reference it here. Maybe even put it in a different module so other collaborators can add other loss functions

Reply:
Good idea, haven't thought of that.
I'll add an extra loss package for autoencoders, which can always be extendet